src/Umbraco.Web.BackOffice/Filters/ContentModelValidator.cs

```diff
@@ -0,0 +1,206 @@
+    /// errors will just be added to the ModelState.
+    /// </remarks>
+    internal abstract class ContentModelValidator<TPersisted, TModelSave, TModelWithProperties>: ContentModelValidator
+        where TPersisted : class, IContentBase
+        where TModelSave: IContentSave<TPersisted>
+        where TModelWithProperties : IContentProperties<ContentPropertyBasic>
+    {
+        private readonly ILocalizedTextService _textService;
+
+        protected ContentModelValidator(ILogger logger, IWebSecurity webSecurity, ILocalizedTextService textService) : base(logger, webSecurity)
+        {
+            _textService = textService ?? throw new ArgumentNullException(nameof(textService));
+        }
+
+        /// <summary>
+        /// Ensure the content exists
+        /// </summary>
+        /// <param name="postedItem"></param>
+        /// <param name="actionContext"></param>
+        /// <returns></returns>
+        public virtual bool ValidateExistingContent(TModelSave postedItem, ActionExecutingContext actionContext)
+        {
+            var persistedContent = postedItem.PersistedContent;
+            if (persistedContent == null)
+            {
+                actionContext.Result = new NotFoundObjectResult("content was not found");
+                return false;
+            }
+
+            return true;
+        }
+
+        /// <summary>
+        /// Ensure all of the ids in the post are valid
+        /// </summary>
+        /// <param name="model"></param>
+        /// <param name="modelWithProperties"></param>
+        /// <param name="actionContext"></param>
+        /// <returns></returns>
+        public virtual bool ValidateProperties(TModelSave model, IContentProperties<ContentPropertyBasic> modelWithProperties, ActionExecutingContext actionContext)
+        {
+            var persistedContent = model.PersistedContent;
+            return ValidateProperties(modelWithProperties.Properties.ToList(), persistedContent.Properties.ToList(), actionContext);
+        }
+
+        /// <summary>
+        /// This validates that all of the posted properties exist on the persisted entity
+        /// </summary>
+        /// <param name="postedProperties"></param>
+        /// <param name="persistedProperties"></param>
+        /// <param name="actionContext"></param>
+        /// <returns></returns>
+        protected bool ValidateProperties(List<ContentPropertyBasic> postedProperties, List<IProperty> persistedProperties, ActionExecutingContext actionContext)
+        {
+            foreach (var p in postedProperties)
+            {
+                if (persistedProperties.Any(property => property.Alias == p.Alias) == false)
+                {
+                    // TODO: Do we return errors here ? If someone deletes a property whilst their editing then should we just
+                    //save the property data that remains? Or inform them they need to reload... not sure. This problem exists currently too i think.
```

Comment:
Interesting, personally I'd like to be notified that it has happened, but not lose data...